lib/streamlit/elements/image_proto.py

```diff
@@ -178,6 +185,8 @@ def marshall_images(
     )
 
     proto_imgs.width = width
+    # Each image in an image list needs to be kept track of at its own coordinates.
+    coord_suffix = 0
```

Comment:
Nit: to avoid possible future bugs (e.g. if someone adds a "continue" inside the loop somewhere), you could use `enumerate()` to handle `coord_suffix` automatically:

```py
for coord_suffix, (image, proto) in enumerate(zip(images, captions)):
    # etc
```